tfhe/src/conformance.rs

```diff
@@ -0,0 +1,22 @@
+/// A trait for objects which can be checked to be conformant with a parameter set
+pub trait ParameterSetConformant {
+    type ParameterSet;
+
+    fn is_conformant(&self, parameter_set: &Self::ParameterSet) -> bool;
+}
+
+/// A constraint on a list size
+/// The list must be composed of a number `n` of groups of size `group_size` which means list size
+/// must be a multiple of `group_size`.
+/// Moreover, `n` must be smaller of equal to `max_group_count`
+#[derive(Copy, Clone)]
+pub struct ListSizeConstraint {
+    pub max_group_count: usize,
+    pub group_size: usize,
+}
+
+impl ListSizeConstraint {
+    pub fn is_valid(&self, size: usize) -> bool {
+        0 < size && size <= self.max_group_count * self.group_size && size % self.group_size == 0
```

Comment:
maybe indicate in the documentation that a list here cannot be empty as we check `0 < size`

Reply:
Adding `min_group_count`